src/DocumentDB.ChangeFeedProcessor/ChangeFeedProcessorBuilder.cs

```diff
@@ -489,6 +501,7 @@ private IPartitionManager BuildPartitionManager(ILeaseStoreManager leaseStoreMan
                     .WithLeaseCollection(this.leaseCollectionLocation)
                     .WithLeaseCollectionLink(collection.SelfLink)
                     .WithRequestOptionsFactory(requestOptionsFactory)
+                    .WithLeaseCollectionPartitionKeyPropertyName(this.leaseCollectionPartitionKeyPropertyName)
```

Comment:
Ideally, this should not be needed. How hard would it be to auto-detect PK definition from lease collection and use it as lease collection PK property name. The only check that would be needed is that PK definition has only 1 property (later it may have more than one).

Reply:
I can fix to auto-detect PK name without asking for user input i.e remove Builder's public method
but partitionkey property name still has to propagate through the internal classes so that we know whether we have to stamp or not. 
other option is to have a global object which is set in this method and then other classes can reference the global object to find out the name.
let me know what you prefer